### PR TITLE
update xmake.lua: more stringent checks on C++20 support for pyincpp

### PR DESCRIPTION
**Pull request by chen-qingyu**
more stringent checks on C++20 support

### CODE REVIEW

**Comment by chen-qingyu** [comment]
咦？怎么windows 2022 也check失败了？看了一下，好像是`configs = {languages = "c++20"}`在`on_check`的`package:check_cxxsnippets`里面没有起作用……

build (windows-2022, shared, arm64, MD):
on check: https://github.com/xmake-io/xmake-repo/actions/runs/9144853292/job/25143144858?pr=4073#step:5:835 
no check: https://github.com/xmake-io/xmake-repo/actions/runs/9144402171/job/25142200595#step:5:1032

**Comment by waruqi** [comment]
先放着吧，我晚上看下

**Comment by waruqi** [comment]
改进了下，应该可以，重跑下试试

File: packages/p/pyincpp/xmake.lua [hunk 1]
```diff
@@ -13,14 +13,24 @@ package("pyincpp")
...
-    -- Some old platforms cannot support the C++20 standard well
+    -- Some old platforms don't support the C++20 standard well.
     if on_check then
-        on_check("android", "macosx", function (package)
+        on_check(function (package)
             if package:version():ge("2.0.0") then
                 assert(package:check_cxxsnippets({test = [[
                     #if !((defined(_MSVC_LANG) && _MSVC_LANG > 201703L) || __cplusplus > 201703L)
                     #error "Require at least C++20."
                     #endif
```
Comment (waruqi) [inline packages/p/pyincpp/xmake.lua:23]: 其实只要有了下面两个特性检测，基本就能过了，尽可能精简下，上面这段不加应该也没事。。然后 error msg 可以放到 assert 里面去

```lua
assert(false, "Require at least C++20.")
```

**Pull request merged**